internal/app/upstream/client_test.go

```diff
@@ -0,0 +1,223 @@
+package upstream_test
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	"github.com/envoyproxy/xds-relay/internal/app/upstream"
+	mock "github.com/envoyproxy/xds-relay/internal/app/upstream/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+type CallOptions = upstream.CallOptions
+
+func TestOpenStreamShouldReturnErrorForInvalidTypeUrl(t *testing.T) {
```

Comment:
You don't have to do this now, but in the future, it might be good to parameterized these tests as you always have to create a MockClient. The only parameters that are changing are CallOptions and the function being passed int.

Reply:
done. There were 2 buckets of unique types of creation across 6 tests and I pulled them into separate functions.